examples/zigzag/main.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2020 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+package main
+
+import (
+	"fmt"
+	"image/color"
+	"log"
+	"math/rand"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+	"github.com/hajimehoshi/ebiten/inpututil"
+)
+
+const (
+	screenWidth  = 640
+	screenHeight = 480
+	Bit32Max     = 0xFFFFFFFF
+)
+
+var (
+	emptyImage *ebiten.Image
+)
+
+// Scale ...
+type Scale struct {
+	X int64
+	Y int64
+}
+
+// Game ...
+type Game struct {
+	moveDirection int8
+	AppleX        int64
+	AppleY        int64
+	scalesLoc     map[int64]*Scale
```

Comment:
I don't understand how scalesLoc works in this program. Could you explain this?

Reply:
scalesLoc is a map. When, snake head collides with apple rectangle. scalesLoc will add a single X,Y position of rectangle, and acts as representation of snake lower body. In Update(), I move the scalesLoc map positions as per snake head movement and direction.